Vostok.Logging.File/Configuration/SafeSettingsCache.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Vostok.Logging.File.Configuration
+{
+    internal class SafeSettingsCache
+    {
+        private static readonly object CooldownGuard = new object();
+
+        private readonly SafeSettingsProvider provider;
+        private readonly TimeSpan ttl = TimeSpan.FromSeconds(1);
+        private readonly bool enabled;
+
+        private volatile object updateCooldown;
+        private volatile Task updateCacheTask = Task.CompletedTask;
+        private volatile FileLogSettings currentSettings;
+
+        public SafeSettingsCache(Func<FileLogSettings> provider)
+        {
+            this.provider = new SafeSettingsProvider(provider);
+
+            var settings = Get();
+
+            enabled = settings.EnableFileLogSettingsCache;
+
+            if (enabled)
+                currentSettings = settings;
+        }
+
+        public FileLogSettings Get()
+        {
+            if (!enabled)
+                return provider.Get();
+
+            RefreshCacheAsync();
+
+            return currentSettings;
+        }
+
+        public void RefreshCache() => RefreshCacheAsync().GetAwaiter().GetResult();
```

Comment:
For #16

Reply:
But this does not really force a refresh.. `RefreshCacheAsync` will do nothing if it encounters a non-null cooldown.

Reply:
This also doesn't seem to propagate the exceptions outward. Am I missing something here, or does the caller not really have a way to tell whether the refresh succeeded?

Reply:
1. Yep, that's a bug :(

2. We use `SafeSettingsProvider` which does not throw exceptions by design. So I supposed that updates are always `succeeded`. You are correct, this should be handled more carefully, because all these caches may become really confusing.
There are actually three objects that may cache settings: `SafeSettingsProvider`, `SafeSettingsCache` and `EventsWriterProvider`. 
The best possible `ForceRefresh` is when you call it and after that your settings are updated everywhere, not only this particular cache. I'll expose it in #16. 
But this method should try updating both it's own cache and `SafeSettingsProvider` cache, throwing exception otherwise.